File: test-template-fix.js
#!/usr/bin/env node
/**
 * Test script to verify the trip templates fix works correctly
 */

import { createClient } from '@supabase/supabase-js';

// Supabase configuration (same as frontend)
const supabaseUrl = 'https://kycoklimpzkyrecbjecn.supabase.co';
const supabaseAnonKey = '<JWT_TOKEN>';

const supabase = createClient(supabaseUrl, supabaseAnonKey);

/**
 * Transform database record to TripTemplate interface (same as service)
 */
function transformDatabaseToTemplate(dbRecord) {
  const templateData = dbRecord.template_data || {};
  
  return {
    id: dbRecord.id,
    name: dbRecord.name,
    description: dbRecord.description || templateData.description || '',
    estimatedDays: templateData.duration_days || templateData.estimatedDays || 7,
    estimatedMiles: templateData.distance_miles || templateData.estimatedMiles || 500,
    difficulty: templateData.difficulty || 'intermediate',
    highlights: templateData.highlights || [],
    suggestedBudget: templateData.estimated_budget || templateData.suggestedBudget || 1000,
    route: templateData.route || null,
    region: templateData.region || 'Rest of the World',
    category: dbRecord.category || 'general',
    tags: dbRecord.tags || [],
    usageCount: dbRecord.usage_count || 0,
    isPublic: dbRecord.is_public || false,
    createdBy: templateData.createdBy
  };
}

/**
 * Test the fixed query for Australian templates
 */
async function testAustralianTemplateQuery() {
  console.log('🧪 Testing Trip Templates Fix');
  console.log('=' * 50);
  
  try {
    // Simulate the fixed query logic for Australia
    const { data, error } = await supabase
      .from('trip_templates')
      .select('*')
      .eq('is_public', true)
      .or(`template_data->>'region'.ilike.*Australia*,template_data->>'region'.ilike.*Victoria*,template_data->>'region'.ilike.*NSW*,template_data->>'region'.ilike.*Queensland*,template_data->>'region'.ilike.*Tasmania*,template_data->>'region'.ilike.*Territory*,tags.cs.{australia}`)
      .order('usage_count', { ascending: false })
      .limit(20);

    if (error) {
      console.error('❌ Query Error:', error);
      return;
    }

    console.log(`✅ Query Success: Found ${data.length} templates`);
    
    if (data.length === 0) {
      console.log('⚠️  No templates found - checking raw data...');
      
      // Check what's actually in the database
      const { data: allData } = await supabase
        .from('trip_templates')
        .select('name, template_data, tags')
        .eq('is_public', true);
        
      console.log('📋 All available templates:');
      allData?.forEach((template, i) => {
        console.log(`  ${i + 1}. ${template.name}`);
        console.log(`     Region: ${template.template_data?.region || 'N/A'}`);
        console.log(`     Tags: ${template.tags?.join(', ') || 'N/A'}`);
      });
      return;
    }

    // Transform and display results
    const transformedTemplates = data.map(transformDatabaseToTemplate);
    
    console.log('\\n🗺️ Australian Trip Templates Found:');
    console.log('=' * 50);
    
    const categories = {};
    transformedTemplates.forEach(template => {
      const category = template.category;
      if (!categories[category]) {
        categories[category] = [];
      }
      categories[category].push(template);
    });

    Object.entries(categories).forEach(([category, templates]) => {
      console.log(`\\n📂 ${category.replace('_', ' ').toUpperCase()}:`);
      templates.forEach(template => {
        console.log(`  • ${template.name}`);
        console.log(`    Duration: ${template.estimatedDays} days | Distance: ${template.estimatedMiles} miles | Budget: $${template.suggestedBudget}`);
        console.log(`    Highlights: ${template.highlights.slice(0, 3).join(', ')}`);
      });
    });

    console.log(`\\n✅ SUCCESS: ${transformedTemplates.length} Australian templates loaded correctly!`);
    console.log('🎯 Expected: 10 templates | Actual: ' + transformedTemplates.length);
    
    if (transformedTemplates.length >= 10) {
      console.log('🎉 TEMPLATE FIX SUCCESSFUL - All templates should now display!');
    } else {
      console.log('⚠️  Some templates may be missing - check database content');
    }

  } catch (error) {
    console.error('❌ Test failed:', error);
  }
}

// Run the test
testAustralianTemplateQuery();

File: backend/test_template_fix.py
#!/usr/bin/env python3
"""
Test the trip template fix to verify Australian templates load correctly
"""

import sys
import os
sys.path.insert(0, os.getcwd())

from app.core.database import get_supabase_client

def test_template_fix():
    print('🧪 Testing Trip Templates Fix')
    print('=' * 50)
    
    # Get database connection
    supabase = get_supabase_client()
    
    try:
        # Get all public templates
        result = supabase.table('trip_templates').select('*').eq('is_public', True).execute()
        
        if not result.data:
            print('❌ No templates found in database')
            return
            
        print(f'✅ Found {len(result.data)} total public templates')
        
        # Filter for Australian templates (simulating frontend logic)
        australian_templates = []
        
        for template in result.data:
            template_data = template.get('template_data', {})
            region = str(template_data.get('region', '')).lower()
            tags = [str(tag).lower() for tag in template.get('tags', [])]
            
            # Check Australian criteria
            is_australian = (
                'australia' in region or
                'victoria' in region or
                'nsw' in region or
                'queensland' in region or
                'tasmania' in region or
                'territory' in region or
                'australia' in tags
            )
            
            if is_australian:
                australian_templates.append(template)
        
        print(f'🗺️ Australian Templates Found: {len(australian_templates)}')
        print()
        
        # Display templates with fixed field mapping
        categories = {}
        for template in australian_templates:
            template_data = template.get('template_data', {})
            category = template.get('category', 'general')
            
            if category not in categories:
                categories[category] = []
                
            # Transform using corrected field mapping
            transformed = {
                'name': template['name'],
                'duration_days': template_data.get('duration_days', 7),
                'distance_miles': template_data.get('distance_miles', 500),
                'estimated_budget': template_data.get('estimated_budget', 1000),
                'region': template_data.get('region', 'Unknown'),
                'category': category,
                'difficulty': template_data.get('difficulty', 'intermediate'),
                'highlights': template_data.get('highlights', [])
            }
            
            categories[category].append(transformed)
        
        # Display by category
        for category, templates in categories.items():
            print(f'📂 {category.replace("_", " ").upper()}:')
            for template in templates:
                print(f'  • {template["name"]}')
                print(f'    Duration: {template["duration_days"]} days | Distance: {template["distance_miles"]} miles | Budget: ${template["estimated_budget"]}')
                print(f'    Difficulty: {template["difficulty"]} | Region: {template["region"]}')
                if template["highlights"]:
                    print(f'    Highlights: {", ".join(template["highlights"][:3])}')
                print()
        
        # Test result
        print('🎯 TEST RESULTS:')
        print(f'Expected: 10+ Australian templates')
        print(f'Found: {len(australian_templates)} templates')
        
        if len(australian_templates) >= 10:
            print('🎉 SUCCESS! Template fix should work correctly')
            print('✅ All Australian RV routes will now display in frontend')
        else:
            print('⚠️  Template count lower than expected')
            print('📝 Check if all templates were inserted correctly')
            
    except Exception as e:
        print(f'❌ Error testing templates: {e}')

if __name__ == "__main__":
    test_template_fix()